CHANGES.md

```diff
@@ -24,6 +24,20 @@ OpenSSL 3.1
 
 ### Changes between 3.0 and 3.1.0 [xx XXX xxxx]
 
+ * Our provider implementations of `OSSL_FUNC_KEYMGMT_EXPORT` and
+   `OSSL_FUNC_KEYMGMT_GET_PARAMS` for EC and SM2 keys now honor
+   `OSSL_PKEY_PARAM_EC_POINT_CONVERSION_FORMAT` as set (and
+   default to `POINT_CONVERSION_UNCOMPRESSED`) when exporting
+   `OSSL_PKEY_PARAM_PUB_KEY`, instead of unconditionally using
+   `POINT_CONVERSION_COMPRESSED` as in previous 3.x releases.
+   For symmetry, our implementation of `EVP_PKEY_ASN1_METHOD->export_to`
+   for legacy EC and SM2 keys is also changed similarly to honor the
+   equivalent conversion format flag as specified in the underlying
+   `EC_KEY` object being exported to a provider, when this function is
+   called through `EVP_PKEY_export()`.
```

Comment:
This talks in terms of providers, but it might be nice to also mention the APIs that are potentially affected for those using the default provider (i.e., `EVP_PKEY_todata`); people might not realise the implications of the above.

Reply:
`EVP_PKEY_todata` is not changing behavior though: you keep getting whatever SECG format a compliant provider decided to use when exporting the public key, exactly as before.

We are only fixing our providers to actually honor the `OSSL_PKEY_PARAM_EC_POINT_CONVERSION_FORMAT` params when serializing `OSSL_PKEY_PARAM_PUB_KEY` (and using _uncompressed_ as default), but even before this change applications should not have expected _compressed_ as the output format on calling `EVP_PKEY_todata` or similar, as that was already potentially incompatible with external providers.

Reply:
Right, I understand this. My point is that a lot of OpenSSL users are only going to be using the default provider and will be implementing their applications by reading manpages without really caring about what provider is used internally. I don't think it would be unreasonable to add something like "Before OpenSSL 3.1, the implementation of these functions in the default provider did not honour `OSSL_PKEY_PARAM_EC_POINT_CONVERSION_FORMAT`, and always returned a compressed key. From OpenSSL 3.1, the default provider now honours this parameter and defaults to returning an uncompressed key."

Reply:
Would 811a7b94c449fe2bbfc9b0c76bc0145b769cc451 suffice?

The problem I see with mentioning that this changed in 3.1, is that soon we might have to alter it (as well as the CHANGES entry), if the upcoming vote to decide if this should be backported to 3.0 passes.

Reply:
Looks good.